### PR TITLE
EVG-5380 add ability to run triggered builds in self-tests

### PR DESCRIPTION
**Pull request by john-m-liu**
also added a make target where you can do
make VENDOR_PKG=github.com/mongodb/grip VENDOR_REVISION=ce377de00786de1277ad2b89194f83c4ed49fa3b revendor

### CODE REVIEW

File: cmd/revendor/run-glide.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		grip.Info("no repo to revendor, run-glide will no-op")
+		return
+	}
+
+	wd, err := os.Getwd()
+	if err != nil {
+		grip.Error(errors.Wrap(err, "error getting working dir"))
+		return
+	}
+	vendorPath := fmt.Sprintf("%s/vendor/%s", wd, pkg)
```
[the remark this answers is not in the document]
Comment (john-m-liu) [reply cmd/revendor/run-glide.go]: fixed

File: cmd/revendor/run-glide.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	_, err = os.Stat(glidePath)
+	if os.IsNotExist(err) {
+		grip.Errorf("glide file %s not found", glidePath)
+		return
+	}
+
+	glideFile, err := os.Open(glidePath)
+	if err != nil {
+		grip.Error(errors.Wrap(err, "error opening glide file"))
+		return
```
[the remark this answers is not in the document]
Comment (john-m-liu) [reply cmd/revendor/run-glide.go]: done

File: cmd/revendor/run-glide.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	"os"
+	"strings"
+
+	"github.com/Masterminds/glide/action"
+	"github.com/Masterminds/glide/repo"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+func main() {
```
[the remark this answers is not in the document]
Comment (john-m-liu) [reply cmd/revendor/run-glide.go]: moved the file handling stuff to its own function

File: cmd/revendor/run-glide.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+		}
+	}
+	if !found {
+		grip.EmergencyFatal("error reading glide file")
+	}
+	err = ioutil.WriteFile(glidePath, []byte(strings.Join(lines, "\n")), 0777)
+	grip.EmergencyFatal(errors.Wrap(err, "error writing glide file"))
+
+	err = os.RemoveAll(vendorPath)
+	grip.EmergencyFatal(errors.Wrap(err, "error removing vendor dir"))
```
Comment (tychoish) [inline cmd/revendor/run-glide.go]: I'd combine 66 and 67 into one line and the same with 69 and 70

File: cmd/revendor/run-glide.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	for i, line := range lines {
+		if strings.Contains(line, pkg) {
+			lines[i+1] = fmt.Sprintf("    version: %s", revision)
+			found = true
+			break
+		}
+	}
+	if !found {
+		grip.EmergencyFatal("error reading glide file")
+	}
```
Comment (tychoish) [inline cmd/revendor/run-glide.go]: I'd omit the found bool, and just call line 64 on line 59.

File: cmd/revendor/run-glide.go [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	action.EnsureGoVendor()
+	action.Install(installer, false, false)
+}
+
+func updateGlide(pkg, revision string) {
+	wd, err := os.Getwd()
+	if err != nil {
+		grip.Error(errors.Wrap(err, "error getting working dir"))
+		return
+	}
```
Comment (tychoish) [inline cmd/revendor/run-glide.go]: is it ok to continue on error here, but not in any other situation?

File: cmd/revendor/run-glide.go (hunk 5 above)
Comment (john-m-liu) [reply cmd/revendor/run-glide.go]: That actually needs to be here, but the error message is wrong. I fixed the message

File: cmd/revendor/run-glide.go (hunk 6 above)
Comment (john-m-liu) [reply cmd/revendor/run-glide.go]: oops, fixed

File: cmd/revendor/run-glide.go (hunk 4 above)
Comment (john-m-liu) [reply cmd/revendor/run-glide.go]: done

File: cmd/revendor/run-glide.go (hunk 5 above)
Comment (tychoish) [reply cmd/revendor/run-glide.go]: needs to be here why?

Comment (john-m-liu) [reply cmd/revendor/run-glide.go]: It's a check to see if the specified package is actually vendored. If it's not I want to exit early

File: cmd/revendor/run-glide.go [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+	action.EnsureGoVendor()
+	action.Install(installer, false, false)
+}
+
+func updateGlide(pkg, revision string) {
+	wd, err := os.Getwd()
+	grip.EmergencyFatal(errors.Wrap(err, "error getting working dir"))
+	vendorPath := filepath.Join(wd, "vendor", pkg)
+	_, err = os.Stat(vendorPath)
+	grip.EmergencyFatal(errors.Wrapf(err, "vendor directory %s not found", vendorPath))
```
Comment (tychoish) [inline cmd/revendor/run-glide.go]: nit: to increase readability, put a blank line after the fatal. 

real: check os.IsNotExist and stat.IsDir() in this situation, the error value from stat is weird.

Comment (john-m-liu) [reply cmd/revendor/run-glide.go]: done

**Pull request merged**